frontend/pages/admin/UserManagementPage/components/UserForm/UserForm.tsx

```diff
@@ -155,17 +196,25 @@ const UserForm = ({
     return () => window.removeEventListener("resize", checkScroll);
   }, [formData]);
 
-  const onInputChange = (formField: string): ((value: string) => void) => {
-    return (value: string) => {
-      setErrors({
-        ...errors,
-        [formField]: null,
-      });
-      setFormData({
-        ...formData,
-        [formField]: value,
-      });
-    };
+  const onInputChange = ({ name, value }: IFormField) => {
+    const newFormData = { ...formData, [name]: value };
+    setFormData(newFormData);
+    const newErrs = validate(newFormData, canUseSso, isNewUser, isSsoEnabled);
+    // only set errors that are updates of existing errors
+    // new errors are only set onBlur or submit
+    const errsToSet: Record<string, string> = {};
+    Object.keys(formErrors).forEach((k) => {
+      // @ts-ignore
+      if (formErrors[k] && newErrs[k]) {
```

Comment:
Additional check, as compared with similar patterns on other forms, for truthiness of the key's value is to handle the unique situation here that we have inherited default error object that does contain keys, e.g. `{email: null}`